### PR TITLE
neovide: initial inclusion in the repository

### PR DESCRIPTION
**Pull request by Achilleshiel**
**Summary**

Fixes: #452

Add Neovide, a simple graphical Neovim client in Rust.

**Test Plan**

- Launched the application
- Edited some text

**Checklist**

- [x] Package was built and tested against unstable

### CODE REVIEW

**Comment by silkeh** [comment]
1. Please wait for a package to be *accepted* before submitting a PR for it.
1. `MAINTAINERS.md` is missing. From [the docs](https://help.getsol.us/docs/packaging/creating-a-new-package):
   > You must add a file called `MAINTAINERS.md` using the template in [Maintainership](https://help.getsol.us/docs/packaging/procedures/maintainership). Solus uses this to track the primary maintainer(s) for each package.

File: packages/n/neovide/files/neovide.desktop
Comment (silkeh) [inline packages/n/neovide/files/neovide.desktop:1]: This file should be included in the sources: https://github.com/neovide/neovide/blob/main/assets/neovide.desktop

File: packages/n/neovide/package.yml
Comment (silkeh) [inline packages/n/neovide/package.yml:11]: `editor` seems more appropriate?

```suggestion
component  : editor
```

File: packages/n/neovide/package.yml
Comment (silkeh) [inline packages/n/neovide/package.yml:12]:
```suggestion
summary    : A simple, no-nonsense, cross-platform graphical user interface for Neovim.
```

File: packages/n/neovide/package.yml
Comment (silkeh) [inline packages/n/neovide/package.yml:14]:
```suggestion
    Neovide is a simple graphical user interface for Neovim (an aggressively refactored and updated Vim editor). 
```

File: packages/n/neovide/package.yml
Comment (silkeh) [inline packages/n/neovide/package.yml:27]: I'm not the most familiar with icons, but why not include the svg?

**Review by silkeh — CHANGES_REQUESTED** [review]
I have gone ahead and accepted neovide to unblock this PR :slightly_smiling_face:. Please add `MAINTAINERS.md` per my previous comments. I also had a few minor comments, nice work!

File: packages/n/neovide/files/neovide.desktop
Comment (ReillyBrogan) [reply packages/n/neovide/files/neovide.desktop:1]: Yes, `Icon=neovide.png` won't work with icon themes. The .desktop file shipped with the source looks fine, just use it:
```
install -Dm00644 assets/neovide.desktop $installdir/usr/share/applications/dev.neovide.Neovide.desktop
```
Note that we're changing the name of the desktop file to the fully qualified one, so this needs a corresponding patch so that window detection works. I went ahead and created this patch for you [here](https://pastebin.com/raw/6BsD2LLL), download it and copy it to the files subfolder and then add `%patch -p1 -i $pkgfiles/0001-Use-fully-qualified-app_id.patch` to your setup phase.

**Review by ReillyBrogan — CHANGES_REQUESTED** [review]

**Review by ReillyBrogan — APPROVED** [review]
LGTM, but FYI it's highly appreciated if on future PRs if you could go through discussions and mark them as "resolved" when you fix them.

**Review by silkeh — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
